javascript/app-framework/src/profiles/renderers/DataTableRenderer.tsx

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { memo } from 'react'
+import type { DataTable } from '@datashaper/workflow'
+import { CompoundEditor } from '../../components/editors/CompoundEditor/index.js'
+import { DataTableEditor } from '../../components/editors/DataTableEditor/index.js'
+import { DataTableTextEditor } from '../../components/editors/DataTableTextEditor/index.js'
+import { ResourceSchemaEditor } from '../../components/editors/ResourceSchemaEditor/index.js'
+import { useHelpOnMount } from '../../hooks/useHelpOnMount.js'
+import type { PluginComponentProps } from '../../types.js'
+
+export const DataTableRenderer: React.FC<PluginComponentProps<DataTable>> =
+	memo(function DataTableRenderer({ resource }) {
+		useHelpOnMount('resources.datatable.index')
+		return <CompoundEditor resource={resource} editors={editors} />
+	})
+
+const editors = [
+	{
+		key: 'datatable-text',
+		title: 'View and edit raw table text content',
+		iconName: 'PageData',
+		renderer: DataTableTextEditor,
```

Comment:
Nice - so these components all just take a `resource` prop?

Reply:
Yerp - as long as it's reactive, the idea is that the component is responsible for figuring out what editing it performs